lib/http.js

```diff
@@ -104,10 +104,68 @@ export async function addRoutes({app, service} = {}) {
         res.status(201).json(body);
       } catch(error) {
         logger.error(error.message, {error});
+        // wrap if not already a BedrockError
+        if(!(error instanceof BedrockError)) {
+          _throwWrappedError({cause: error});
+        }
         throw error;
       }
 
       // meter operation usage
       metering.reportOperationUsage({req});
     }));
 }
+
+// known error cause names to map to 'DataError'
+const _dataErrors = new Set([
+  'DataError',
+  'jsonld.InvalidUrl',
+  'jsonld.ValidationError'
+]);
+// TODO: handle other possible errors
+// in particular @digitalbazaar/vc throws many simple 'TypeError' and
+// 'Error' errors. Difficult to distinguish them from other errors.
+
+function _throwWrappedError({cause}) {
+  const {name/*, message*/} = cause;
+
+  // TODO: should this be verbose with the top level error?  it's also in the
+  // 'error' field.
+  //const error = new BedrockError(message ?? 'Invalid credential.', {
+  const error = new BedrockError('Invalid credential.', {
+    name: _dataErrors.has(name) ? 'DataError' : 'OperationError',
+    details: {
+      // using sanitized 'error' field instead of 'cause' due to bedrock
+      // currently filtering out non-BedrockError causes.
+      error: _stripStackTrace(cause),
+      httpStatusCode: cause.httpStatusCode ?? 400,
+      public: true
+    }
+  });
+
+  throw error;
+}
+
+function _stripStackTrace(error) {
```

Comment:
This implementation needs to be updated (it was taken from a source where some bugs were discovered):

https://github.com/digitalbazaar/bedrock-vc-delivery/blob/main/lib/helpers.js#L151-L170

Once that's updated, I think this is ready to go.

Reply:
Aligned more with that code.  jsonld.js doc loader is using `.details.cause` for not found errors, which also need stripping. Not sure if that's an issue in the other library as well.

Reply:
I think it's probably fine.